dialogue-java-client/src/main/java/com/palantir/dialogue/HttpChannel.java

```diff
@@ -110,7 +112,16 @@ private Response toResponse(HttpResponse<InputStream> response) {
         return new Response() {
             @Override
             public InputStream body() {
-                return response.body();
+                boolean isGzipped = response.headers()
+                        .firstValue("content-encoding").orElse("")
+                        .toLowerCase()
+                        .equals("gzip");
+                if (isGzipped) {
+                    // TODO(rfink): Think about removing the okio dependency at some point?
```

Comment:
why do we need Okio here?

Reply:
maybe we should use the Java built-in https://docs.oracle.com/javase/8/docs/api/java/util/zip/GZIPInputStream.html?
```java
return new GZIPInputStream(response.body());
```

Reply:
That thing unfortunately reads the GZIP header in the constructor and thus blocks the otherwise async wiring.